api/config/v1alpha1/shared_types.go

```diff
@@ -10,6 +10,14 @@ const (
 	DefaultEnvoyReplicas = 1
 )
 
+// GroupVersionKind unambiguously identifies a Kind.
+// It can be converted to k8s.io/apimachinery/pkg/runtime/schema.GroupVersionKind
+type GroupVersionKind struct {
```

Comment:
can this be reused from upstream ?

Reply:
The main addition here is that it is JSON serializable which the upstream version afaik isn't. 
And I don't think Go always adding additional methods (like a `MarshallJSON` method) to types from other packages.